glmmbernoulli.qmd

```diff
@@ -0,0 +1,523 @@
+```
+
+producing the population predictions on the probability scale, as shown in @fig-com05predmu.
+
+```{julia}
+#| code-fold: true
+#| fig-cap: Predicted probability of contraception use versus centered age from model com05.
+#| label: fig-com05predmu
+draw(
+  data(newdata) *
+  mapping(
+    :age => "Centered age (yr)",
+    :μ => "Probability of contraceptive use";
+    col=:urban => renamer(["N" => "Rural", "Y" => "Urban"]),
+    color=:ch => "Children",
+  ) *
+  smooth();
+  figure=(; resolution=(800, 450)),
+)
+```
+
+On the probability scale we can compare the predictions to the observed frequencies shown with scatterplot smoother lines in @fig-contradata2.
+
+Consider the predictions on both the linear predictor and probability (or expected value) scales for women with centered ages of 2.0.
+
+```{julia}
+filter(r -> r.age == 2, newdata)
+```
+
+The predicted probability of woman with centered age of 2, with children, living in an urban environment using artificial contraception is about 2/3, which is reasonably close to the smoothed frequency for that combination of covariates in @fig-contradata2.
+
+Similarly, a woman of centered age of 2 without children living in a rural environment has a predicted probability of using artificial contraception of a little less than 20%, which also corresponds to the smoother line for that combination (blue line in the left panel) in @fig-contradata2.
+
+## Interpretation of random effects {#sec-glmmrandomeff}
+
+We should also be aware that the random effects are defined on the linear predictor scale and not on the probability scale.
+A normal probability plot of the conditional modes of the random effects for model `com05`, @fig-com05qqcaterpillar
+
+```{julia}
+#| code-fold: true
+#| fig-cap: Caterpillar plot of the conditional modes of the random-effects for model `com05`
+#| label: fig-com05qqcaterpillar
+qqcaterpillar(com05)
+```
+
+shows that the smallest random effects are approximately -1 and the largest are approximately 1.
+
+The numerical values and the identifier of the combination of `dist` and `urban` for these extreme values can be obtained from the first few rows and the last few rows of the sorted random-effects table.
+
+```{julia}
+srtdre = let retbl = only(raneftables(com05))
+  retbl[sortperm(last.(retbl))]
+end
+first(srtdre, 6)
+```
+
+```{julia}
+last(srtdre, 6)
+```
+
+The largest random effect is for rural settings in `D34`.
+There were 26 women in the sample from rural `D34`
+
+```{julia}
+D34N = filter(r -> r.dist == "D34" && r.urban == "N", contra)
+```
+
+of whom 20 used contraception, an unusually large proportion for a rural setting.
+
+But this happens when you have relatively small survey sizes - we expect considerable variation.
+
+Also there is considerable variability in the lengths of the prediction intervals in @fig-com05qqcaterpillar because the data are unbalanced with respect to district and rural/urban districts.
```

Comment:
Should we add a feature to MixedModelsMakie to allow plotting random effects on the response scale? (Well, I guess at that point, we might as well add that option for the fixed effects, but that seems like a mistake.)

Reply:
Using the response scale for coefficients, either fixed effects or random effects, is tricky because you can't isolate the effect of an individual coefficient on mu.  I.e. an effect of 0.1 can be substantial when the other components give eta = -0.05 but is negligible when the other components give eta = 4.5

I made one plot on the mean scale showing the predictions for a "typical" district/urban combination and according to the fixed effects.  One of the purposes of that plot is to show that the quadratic pattern for a particular combination of `ch` and `urban` in eta is not preserved on the mu scale.   Should that be emphasized more in the text?

Reply:
No, I think the text is fine. I was just thinking out loud and agree that it gets very tricky very fast and we would be making a potential footgun for a lot of users.